packages/flutter_tools/lib/src/ios/ios_emulators.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../base/process.dart';
+import '../emulator.dart';
+import '../globals.dart';
+import 'ios_workflow.dart';
+
+class IOSEmulators extends EmulatorDiscovery {
```

Comment:
nit: technically, iOS doesn't provide emulators, just simulators that run x86_64 code on a faked-out platform.

Reply:
Yeah; this naming is something we couldn't really settle on. We already have a bunch of classes using the name "Simulator" which are used in the `devices` code (which represent active running simulators). The `emulators` section lists emulators/simulators that are not necessarily running. Naming these ones Simulators will cause confusion (or collisions) with the other classes (the ones that represent online simulators).

I'm happy to rename, but we couldn't come up with something more accurate that might not also be confusing. (I guess the same comment applies to the user-facing `flutter emulators` command too).

Reply:
I'll be honest, I'm pretty stumped too. SimulatedDevice sounds too close to Simulator and technically it's an EmulatedDevice for Android. SoftwareDevice is pretty cumbersome and I don't think I've ever heard anyone use that term either way :) `IOSEmulators` seems like a reasonable compromise.